include/GafferUI/Gadget.h

```diff
@@ -341,6 +354,8 @@ class Gadget : public Gaffer::GraphComponent
 typedef Gaffer::FilteredChildIterator<Gaffer::TypePredicate<Gadget> > GadgetIterator;
 typedef Gaffer::FilteredRecursiveChildIterator<Gaffer::TypePredicate<Gadget> > RecursiveGadgetIterator;
 
+inline Gadget::Layer &operator++( Gadget::Layer &x ) { return x = (Gadget::Layer)(((int)(x) + 1)); };
```

Comment:
Potentially superfluous if we don't do the `auto layer; ++layer` gubbins anymore?